agents/sorenson_dlm/dlm_agent.py

```diff
@@ -0,0 +1,259 @@
+        self.send_msg('STAT:PROT:ENABLE?')
+        enb = self.rec_msg()
+        if enb != '8':
+            print('Error: Over voltage protection failed to enable')
+            return
+
+        self.send_msg('STAT:PROT:EVENT?')
+        event = self.rec_msg()
+        if event != '0':
+            print('Error: Over voltage already tripped')
+            return
+
+    def read_voltage(self):
+        """
+        Reads output  voltage
+        """
+        self.send_msg('SOUR:VOLT?')
+        msg = self.rec_msg()
+        return msg
+
+    def read_current(self):
+        """
+        Reads output current
+        """
+        self.send_msg('MEAS:CURR?')
+        msg = self.rec_msg()
+        return msg
+
+    def sys_err_check(self):
+        """
+        Queries sytem error and returns error byte
+        """
+        self.send_msg('SYST:ERR?')
+        msg = self.rec_msg()
+        return msg
+
+
+class DLMAgent:
+    '''
+    TO DO:
+    set_current_and_voltage (simulatneously set current and voltage,
+    not sure if needed)
+
+    '''
+
+    def __init__(self, agent, ip_address , port, f_sample=2.5):
+        self.active = True
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.f_sample = f_sample
+        self.take_data = False
+        self.over_volt = 0
+        self.dlm = DLM(ip_address, int(port))
+        agg_params = {'frame length': 60, }
+        self.agent.register_feed('voltages',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1)
+
+    def start_acq(self, session, params=None):
+        '''
+        Get voltage and current values from the sorenson, publishes them to
+        the feed
+
+        Args:
+            sampling_frequency: defaults to 2.5Hz
+        '''
+        if params is None:
+            params = {}
+
+        f_sample = params.get('sampling_frequency')
+        if f_sample is None:
+            f_sample = self.f_sample
+
+        sleep_time = 1./f_sample - 0.01
+        with self.lock.acquire_timeout(timeout=0, job='init') as acquired:
+            # Locking mechanism stops code from proceeding if no lock acquired
+            if not acquired:
+                self.log.warn("Could not start init because {} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            self.take_data = True
+
+            while self.take_data:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'voltages',
+                    'data': {}
+                }
+                voltage_reading = self.dlm.read_voltage()
+                current_reading = self.dlm.read_current()
+                print('Voltage: {}'.format(voltage_reading))
+                print('Current: {}'.format(current_reading))
```

Comment:
Useful for debugging, but we don't want to always print this. If you'd like to keep it around, please change the print's to `self.log.debug()` instead.